metrics/docs/02_sign_and_notarize_metrics_binaries.md

```diff
@@ -0,0 +1,176 @@
+ - a debug can be enabled with `-log-json` flag
+ - can be run with `notarization-only` configuration
+ - notarize packages and wait for the notarization to complete
+ - signed files into a `.dmg` or `.zip`
+ - popular tool
+
+Cons:
+ - can't be used with `sign-only` configuration
+
+### [signing_tools](https://github.com/drud/signing_tools)
+
+This is two `.sh` scripts that provides sign and notarize binaries.
+
+First, `macos_sign.sh` is the sign script. It accepts the following list of parameters: 
+ - `--signing-password` - a password for key-chain, that the tool is creating while running;
+ - `--cert-file` - a developer id application certificate, used for code sign
+ - `--cert-name` - a name of the certificate
+ - `--target-binary` - a specific binary
+
+Example of usage:
+
+```
+./macos_sign.sh --signing-password="${SIGNING_TOOLS_SIGNING_PASSWORD}" --cert-file=${CERTFILE} --cert-name="${CERTNAME}" --target-binary="${TARGET_BINARY}"
+```
+
+Second, `macos_notarize.sh` is the notarize script. It accepts the following list of parameter:
+ - `--app-specific-passwords` - the generated password for the application
+ - `--apple-id` - an id of the apple developer account
+ - `--primary-bundle-id`
+ - `--target-binary` - a specific binary
+
+ ```
+ ./macos_notarize.sh --app-specific-password=${APP_SPECIFIC_PASSWORD} --apple-id=${APPLE_ID} --primary-bundle-id=com.ddev.test-signing-tools --target-binary=${TARGET_BINARY}
+ ```
+
+Pros:
+ - separate scripts for sign and notarize step, so the tool can be used just for sign or notarize only
+ - notarize step checks if the given binary is already signed
+ - notarize step wait until notarization on the Apple's side completes
+ - logs issues during the process
+
+Cons:
+ - the script's bash syntax is not easy to understand
+ - does not have a single configuration file
+ - not popular tool
+
+### [notarize-cli](https://github.com/bacongravy/notarize-cli)
+
+This tool is a wrapper for `xcrun altool` and `xcrun stapler`, so it requires Xcode to be installed.
+
+There is a list of parameters:
+  - `--bundle-id` - bundle id of the app to notarize (can pass via ENV)
+  - `--file` - path to the file to notarize
+  - `--password` - password to use for authentication (can pass via ENV)
+  - `--username` - username to use for authentication (can pass via ENV)
+
+Example of usage:
+
+```
+npx notarize-cli [OPTIONS]
+```
+
+The tool is just for notarizing we need to sign from within a separate step manually. 
+
+To do so we can:
+ 1. use Github Actions (https://github.com/Apple-Actions/import-codesign-certs) to import code-sign certificates
+ 2. sign binary using the following command: `codesign --force -s <identity-id> ./path/to/you/app -v`
+
+Pros:
+ - installation is unnecessary via npx
+ - wait until notarization on the Apple's side completes
+ 
+Cons:
+ - only for notarize
+ - requires xcode
+ - does not check if the given file is already signed
+
+## Decision
```

Comment:
I think we also need to mention the blocker that we found here. So the decision would be that due to the blocker go with sha256 approach until blocker is solved. If blocker is fixed then we would use gon.

That means that in the analysis phase we would want to add sha256 approach with pros/cons too.